### PR TITLE
detect-icmp-seq: convert unittests to FAIL/PASS APIs

### PR DESCRIPTION
**Pull request by AkakiAlice**
Task: #4043

Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/4043

Describe changes:
- Update detect-icmp-seq to use FAIL/PASS API

### CODE REVIEW

**Comment by AkakiAlice** [comment]
I see that the ``DetectIcmpSeqMatchTest01`` test present in this file is a good candidate to be converted to a Suricata-Verify test. I would like to work on it, should I create a task?

**Comment by jufajardini** [comment]
> I see the `DetectIcmpSeqMatchTest01` test present in this file as a good candidate to be converted to a Suricata-Verify test. I would like to work on it, should I create a task?

If there isn't one yet, yes, that's a good idea :)

File: src/detect-icmp-seq.c [hunk 1]
```diff
@@ -352,11 +350,9 @@ static int DetectIcmpSeqParseTest02 (void)
 {
     DetectIcmpSeqData *iseq = NULL;
     iseq = DetectIcmpSeqParse(NULL, "  300  ");
-    if (iseq != NULL && htons(iseq->seq) == 300) {
-        DetectIcmpSeqFree(NULL, iseq);
-        return 1;
-    }
-    return 0;
+    FAIL_IF_NOT(iseq != NULL && htons(iseq->seq) == 300);
```
Comment (victorjulien) [inline src/detect-icmp-seq.c:353]: same

File: src/detect-icmp-seq.c [hunk 2]
```diff
@@ -366,11 +362,9 @@ static int DetectIcmpSeqParseTest03 (void)
 {
     DetectIcmpSeqData *iseq = NULL;
     iseq = DetectIcmpSeqParse(NULL, "badc");
-    if (iseq != NULL) {
-        DetectIcmpSeqFree(NULL, iseq);
-        return 0;
-    }
-    return 1;
+    FAIL_IF_NOT_NULL(iseq);
+    DetectIcmpSeqFree(NULL, iseq);
```
Comment (victorjulien) [inline src/detect-icmp-seq.c:366]: useless statement? It will only be called with `iseq == NULL`

File: src/detect-icmp-seq.c [hunk 3]
```diff
@@ -366,11 +362,9 @@ static int DetectIcmpSeqParseTest03 (void)
 {
     DetectIcmpSeqData *iseq = NULL;
```
Comment (victorjulien) [inline src/detect-icmp-seq.c:363]: in your next PR, could you change these to a single line, e.g.
`DetectIcmpSeqData *iseq = DetectIcmpSeqParse(NULL, "badc");`

**Comment by AkakiAlice** [comment]
> see inline comments

Thanks for the feedback, I will make the changes

**Pull request closed without merging**